### PR TITLE
GEOMESA-824 Aggregating bin records in tablet servers

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Adding optional sorting of bin records, since regular feature sorting will no longer work
* BinaryViewerOutputFormat expands aggregated records
* Non-z3 indices do not aggregate yet

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala [hunk 1]
```diff
@@ -180,6 +183,22 @@ object RichAttributeDescriptors {
...
 }
 
+object RichSimpleFeatureType {
+
+  import RichAttributeDescriptors.RichAttributeDescriptor
+  import SimpleFeatureTypes.DEFAULT_DATE_FIELD
+
+  import scala.collection.JavaConversions._
+
+  implicit class RichSimpleFeatureType(val sft: SimpleFeatureType) extends AnyVal {
```
Comment (mmatz-ccri) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala]: merge with caution: in master there already is an "implicit class RichSimpleFeatureType" as a nested class of "object Conversions"

Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala]: ah, thanks. i need to rebase this on master, actually.

Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala]: rebased and fixed.

**Comment by anthonyccri** [comment]
Merged

**Pull request closed without merging**